src/transaction-builder.ts

```diff
@@ -43,6 +47,17 @@ export class TransactionBuilder {
     beneficiary?: Address;
     onlyParams?: boolean;
   }) {
+    if (deadline) {
+      const globalDeadline = +deadline;
+      if (!isNaN(globalDeadline)) {
```

Comment:
What if `globalDeadline` isNan? I think in that case it will continue execution, which may be not right

Reply:
I agree better fire an error